### ISSUE TITLE
does not work on Debian/OpenMediaVault (it uses dash)

### ISSUE THREAD

**Issue by the-Arioch** [gamestailer94/farm-check#24]
was trying to run it in my "unused garbage made NAS" and it was telling me plain wrong error about having no smartmon installed!

the problem block is

```
if ! smartctl -V &> /dev/null; then
    echo "Error: smartctl not found. Please install smartmontools."
    exit 1
fi
```

run like charm in interactive bash but not in interactive sh

i am a typical Windows mouse mover, so i feared to just change shebang to bash...

> it would be nice if these was an explicit claim about it, if this script is safe to be bash-forced or if bash mode would break it.

i thought it was some busybox incompatibility, but `ls -la` told me `sh` was some `dash` not busybox (should there be a punishment to whoever writes one more POSIX shell there? ir is it too late?)

Google failed me to teach me much about dash (so perhaps could not engineer good enough prompt) but at least lead me to https://dev.to/bowmanjd/writing-bash-scripts-that-are-not-only-bash-checking-for-bashisms-and-testing-with-dash-1bli

this article led me to https://www.shellcheck.net and it is not happy

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/dash

if ! smartctl -V &> /dev/null; then
    echo "Error: smartctl not found. Please install smartmontools."
    exit 1
fi
```

#### Here's what shellcheck currently says:
```
Line 3	SC3020: In dash, &> is not supported.
```

I took an advice from https://www.shellcheck.net/wiki/SC3020 and after that your script looks working, though i am not quite sure

**Comment by the-Arioch** [comment]
P.S. this made me giggle

```
=== Checking device: /dev/sde ===
Model Family:   N/A (smartmontools does not know this device or device does not report Model Family)
Device Model:   ST8000VX010-2ZR188
Serial Number:  WPV20PB1

SMART:          4
FARM:           4
HEAD:           PASS (Max: 0 hrs, Min: 999999999 hrs)
RESULT:         PASS

=== Checking device: /dev/sdf ===
Model Family:   Seagate Desktop HDD.15
Device Model:   ST4000DM000-1F2168
Serial Number:  W300E3CQ

                FARM data not available - likely not a Seagate drive
SMART:          45390
FARM:           N/A
HEAD:           N/A
RESULT:         SKIP
```

Are ex-Maxtor disks out of water completely, because i think Seagate still sells some xxDMxx models? Or maybe some script modifications could find required API in them?

**Comment by derhuerst** [comment]
I'm running into this problem too, with Ubuntu 25, with `smartmontools` installed.

```shell
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 25.04
Release:	25.04
Codename:	plucky
```

**Comment by Parlane** [comment]
On debian change #!/bin/sh to #!/bin/bash and the script works correctly. Should work fine on ubuntu too.

**Comment by the-Arioch** [comment]
or just remove one unneeded bashism from the script - that should be
potentially less destructive, who knows how many behavior differences might
global engine switch trigger

чт, 28 авг. 2025 г. в 00:26, Matthew Parlane ***@***.***>:
